path.go

```diff
@@ -9,7 +9,7 @@ import (
 )
 
 func zoneFromPath(host string, path string, rec record) (string, int, error) {
-	match, err := regexp.Compile("([a-zA-Z0-9]+)")
+	match, err := regexp.Compile("[^\\/#\\.]+")
```

Comment:
What's the specific reasoning behind this change? Was personally thinking about a different regex. `([^/]*)`

Reply:
I've used this regex because browsers don't support # in URL if they are not at the end before a character. But I'm not sure about it. we can use the one you suggested if what I said isn't right.

Reply:
So for a string like: `test/123#something`
Your regex would match 3 times: test, 123, something
The more general regex would just match 2 times: test, 123#something

Question would be, what would we expect?
In my view both seem slightly wrong. What we want would be `test`, `123` and then add the additional fragment to the resulting redirect, if enabled right? At least that would be my guess within a default one. Basically just use the path nothing more. Using fragments or query parameters could then be done via custom regex.

Reply:
The only solution that I can think of right now is to split the path string into 2 parts from # and apply the regex only on the first part.

Reply:
Going through this in my head. It seems we have to choices.

Only act on the path and therefore the regex only cares about the path.
Or we use a different regex. Let me check, if I can come up with one.

Reply:
`\/([A-Za-z0-9-._~!$&'()*+,;=:@]+)` assuming I got the allowed chars right for paths.

We can then use grouping.
So the match could be /test, but the group we are actually interested in would be the following item in the given array.

Example: https://play.golang.org/p/slASFcOceO5

That way we get the match and the thing we actually want.

Does that make sense or am I missing something?